frontend/src/components/Toast.jsx

```diff
@@ -0,0 +1,54 @@
+import { CheckCircle, XCircle, X, ExternalLink } from 'lucide-react'
+import { useToast } from '../contexts/ToastContext'
+import '../styles/Toast.css'
+
+function Toast({ toast }) {
+  const { removeToast } = useToast()
+  const { id, type, title, message, txHash, explorerUrl } = toast
+
+  const Icon = type === 'success' ? CheckCircle : XCircle
+
+  const handleExplorerClick = () => {
+    if (explorerUrl && txHash) {
+      window.open(`${explorerUrl}/tx/${txHash}`, '_blank', 'noopener,noreferrer')
+    }
+  }
+
+  return (
+    <div className={`toast toast-${type}`}>
+      <div className="toast-icon">
+        <Icon size={24} />
+      </div>
+      <div className="toast-content">
+        {title && <div className="toast-title">{title}</div>}
+        <div className="toast-message">{message}</div>
+        {txHash && explorerUrl && (
+          <button className="toast-explorer-link" onClick={handleExplorerClick}>
+            <span>View on Explorer</span>
+            <ExternalLink size={14} />
```

Comment:
I think that using external link is good enough for this? 

```suggestion
            <ExternalLink size={14} />
```